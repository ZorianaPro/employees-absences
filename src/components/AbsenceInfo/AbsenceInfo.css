.Absence-Info {
  padding-left: 0.5em;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.Absence-Info.Absence-Info--isLinesSeparated {
  padding: 0.5em 0;
  box-shadow: 0 3px 0 -2px rgba(21, 22, 26, 0.15);
}