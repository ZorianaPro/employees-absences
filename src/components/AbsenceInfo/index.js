import React from 'react';
import './AbsenceInfo.css';

const AbsenceInfo = ({
  id,
  type,
  startDate,
  endDate,
  view
}) => (
  <li className="Absence-Info"
    key={ `absenceInfo-${id}` }>
    <p>is on { type }
      { view === 'day'
        ? ' today'
        : ` from ${startDate} to ${endDate}` }</p>
  </li>
);

export default AbsenceInfo;