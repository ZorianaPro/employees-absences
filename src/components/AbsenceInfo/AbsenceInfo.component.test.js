import React from 'react';
import AbsenceInfo from './';
import { shallow, mount } from 'enzyme';

const data = {
  'id': 1,
  'type': 'vacation',
  'startDate': '2020-01-13',
  'endDate': '2020-02-13',
  'view': 'month'
};

describe('AbsenceInfo', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(
      shallow,
      <AbsenceInfo
        id={ data.id }
        type={ data.type}
        startDate={ data.startDate}
        endDate={ data.endDate }
        view={ data.view }
      />
    )).not.toThrow();
  });

  it('renders as expected', () => {
    expect(mount(
      <AbsenceInfo
        id={ data.id }
        type={ data.type}
        startDate={ data.startDate}
        endDate={ data.endDate }
        view={ data.view }
      />
    )).toMatchSnapshot();
  });

  describe('view', () =>  {
    describe('is day', () =>  {
      let absenceInfo;
      beforeEach(() => {
        absenceInfo = mount(
          <AbsenceInfo
            id={ data.id }
            type={ data.type}
            startDate={ data.startDate}
            endDate={ data.endDate }
            view='day'
          />
        );
      });

      it('renders absence with today word', () => {
        expect(absenceInfo.find('.Absence-Info .Absence-Info-Date').text())
          .toEqual('today');
      });
    });

    describe('is month', () =>  {
      let absenceInfo;
      beforeEach(() => {
        absenceInfo = mount(
          <AbsenceInfo
            id={ data.id }
            type={ data.type}
            startDate={ data.startDate}
            endDate={ data.endDate }
            view='month'
          />
        );
      });

      it('renders absence with range', () => {
        expect(absenceInfo.find('.Absence-Info .Absence-Info-Date').text())
          .toEqual('from 2020-01-13 to 2020-02-13');
      });
    });
  });
});