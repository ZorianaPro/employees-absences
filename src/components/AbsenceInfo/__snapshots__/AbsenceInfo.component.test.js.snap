// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbsenceInfo renders as expected 1`] = `
<AbsenceInfo
  endDate="2020-02-13"
  id={1}
  startDate="2020-01-13"
  type="vacation"
  view="month"
>
  <li
    className="Absence-Info"
    key="absenceInfo-1"
  >
    <p>
      is on 
      vacation
       from 2020-01-13 to 2020-02-13
    </p>
  </li>
</AbsenceInfo>
`;
